### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by itye-msft**
After writing multiple integration tests which involved communication with event hub, I ended up removing them and sticking to integration in an isolated environment.
The full e-2-e test with connectivity to event hub will be added in a  separated PR

### CODE REVIEW

File: agogosml/tests/integration_tests/test_integ_input_app_output.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    print('Creating writer')
+    ow_streaming_client = ClientMessagingMock()
+    ow = OutputWriterFactory.create(ow_config, ow_streaming_client, None)
+    print('start_incoming_messages')
+    ow.start_incoming_messages()
+
+    print('sending test message to reader')
+    test_msg = 'test'
+    ir_streaming_client.mock_incoming_message_event(test_msg)
+    last_msg = ow_streaming_client.get_last_msg()
```
Comment (fnocera) [inline agogosml/tests/integration_tests/test_integ_input_app_output.py]: `self.last_message` is never set in the `ClientMessagingMock()` for the output_writer therefore when `last_msg = ow_streaming_client.get_last_msg()` remains `None`

File: agogosml/tests/integration_tests/test_integ_input_app_output.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+@pytest.fixture
+def mock_streaming_client():
+    return ClientMessagingMock()
+
+
+@pytest.fixture
+def mock_listener_client():
+    return ListenerClientMock(0)
+
+
```
Comment (fnocera) [inline agogosml/tests/integration_tests/test_integ_input_app_output.py]: should this have `@pytest.mark.integration` above the function?

File: agogosml/tests/integration_tests/test_integ_input_app_output.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+def test_when_messages_received_in_input_then_all_messages_are_sent_via_output():
```
Comment (fnocera) [inline agogosml/tests/integration_tests/test_integ_input_app_output.py]: should this have `@pytest.mark.integration` above the function?

File: agogosml/tests/test_it_output_writer.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+from dotenv import load_dotenv
+from agogosml.writer.output_writer import OutputWriter
+from agogosml.writer.output_writer_factory import OutputWriterFactory
+from agogosml.common.flask_http_listener_client import FlaskHttpListenerClient
+from agogosml.common.kafka_streaming_client import KafkaStreamingClient
+from agogosml.common.eventhub_streaming_client import EventHubStreamingClient
+
+
+
+def test_when_new_message_then_send_to_event_hub():
```
Comment (fnocera) [inline agogosml/tests/test_it_output_writer.py]: should this have `@pytest.mark.integration` above the function?

File: agogosml/agogosml/reader/input_reader_factory.py [hunk 5]
```diff
@@ -13,9 +13,10 @@ class InputReaderFactory:
     """Factory and instance resolving for input reader"""
 
     @staticmethod
-    def create(config: dict):
+    def create(config: dict, streaming_client=None):
         """Create a new instance of InputReader
 
+        :param streaming_client:
```
Comment (torosent) [inline agogosml/agogosml/reader/input_reader_factory.py]: The order of the params is reversed to the signature. Please add more details in the description

**Pull request merged**